code/game/objects/structures/holosign.dm

```diff
@@ -104,6 +102,70 @@
 	. = ..()
 	air_update_turf(TRUE)
 
+///this is a machinery to make it possible to check on the APC power to shut it down in case of an outage
+/obj/machinery/holosign/barrier/power_shield
+	name = "powered shield"
+	desc = "A shield to prevent changes of atmospheric and heat transfer"
+	icon = 'icons/effects/effects.dmi'
+	density = FALSE
+	anchored = TRUE
+	CanAtmosPass = ATMOS_PASS_NO
+	resistance_flags = FIRE_PROOF
+	///store the conductivity value of the turf is applyed so that it can be restored on removal
+	var/stored_conductivity = 0
+	///power drain from the apc, in W (so 5000 is 5 kW), per each holosign placed
+	var/power_consumption = 5000
+	var/obj/item/holosign_creator/shield_projector
+
+/obj/machinery/holosign/barrier/power_shield/Initialize(loc, source_projector)
+	. = ..()
+	var/area/a = get_area(src)
+	if(a.power_equip == FALSE)
+		stack_trace("Power shield created without power avaiable")
+		return INITIALIZE_HINT_QDEL
+	if(source_projector)
+		shield_projector = source_projector
+		LAZYADD(shield_projector.signs, src)
+		air_update_turf(TRUE)
+		a.addStaticPower(power_consumption, AREA_USAGE_STATIC_EQUIP)
+		shield_turf()
+
+/obj/machinery/holosign/barrier/power_shield/Destroy()
+	var/turf/T = loc
+	T.thermal_conductivity = stored_conductivity
+	var/area/a = get_area(src)
+	a.addStaticPower(-power_consumption, AREA_USAGE_STATIC_EQUIP)
+	LAZYREMOVE(shield_projector.signs, src)
+	shield_projector = null
+	return ..()
+
+///Proc that takes the thermal conductivity of the turf its on and store it inside a variable
+/obj/machinery/holosign/barrier/power_shield/proc/shield_turf()
+	var/turf/T = loc
+	if(isturf(loc))
+		stored_conductivity = T.thermal_conductivity
+		T.thermal_conductivity = 0
+
+/obj/machinery/holosign/barrier/power_shield/process()
+	if(machine_stat & NOPOWER)
+		qdel(src)
```

Comment:
Could you turn this into an event instead? To get deleted when it loses power, and not to process by default?